### PR TITLE
Add option to adjust lightning fee limit

### LINKED ISSUE #121: Zap fails to pay "private" LN invoices
I can't pay invoices created from the breez wallet. Their nodes are private. Essentially, Zap for android can't pay any invoice for which the fee is unknown. It says no path to destination. The same invoice can be paid no problem with the desktop version of Zap for windows, connected to the same LND node.

- Zap version: (e.g. 0.2.5-alpha)
- Device: (Samsung Galaxy S8)
- LND Version: ( 0.7.1-beta)

### PR DESCRIPTION
**Pull request by maraph**
## Description
1. Allow user to set a fee limit for lightning payments.
2. Use mission control for query routes
3. Fix #121 

## Motivation and Context
Fix bug where payments could not be routed to private channels.
Allow user to define a limit for lightning fees.

## How Has This Been Tested?
Testnet, Mainnet

## Screenshots (if appropriate):

payment_overview

dialog_alert

setting_overview

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [Contribution document](../docs/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Relaxo143** [comment]
The idea for fee limits is great! However there should be an option for the user to set the limits directly in sats. If the invoice is for 10 sats for example and the fee is 1 sat, a warning would be triggered since that's 10% of the amount.

**Comment by maraph** [comment]
@Relaxo143 We would only use the % limit for payments above a certain threshold, which currently is above 100 sats. We probably could make this more prominent or let the user decide the threshold.

**Comment by Relaxo143** [comment]
> @Relaxo143 We would only use the % limit for payments above a certain threshold, which currently is above 100 sats. We probably could make this more prominent or let the user decide the threshold.

Sounds good!

**Comment by maraph** [comment]
@Relaxo143 So now we would show the threshold in the settings page. 
Maybe we will make this adjustable in the future.

settings_overview

File: app/src/main/java/zapsolutions/zap/fragments/SendBSDFragment.java [hunk 1]
```diff
@@ -504,85 +511,50 @@ public void onClick(View v) {
...
+                                            .build();
+                                }
+                            } else {
+                                // fee higher than payment
+                                if (mLnFeePercentCalculated >= 1f) {
+                                    int feeSats = (int) (mLnPaymentRequest.getNumSatoshis() * mLnFeePercentCalculated);
+                                    String feeLimitString = getString(R.string.fee_limit_exceeded_payment, feeSats, mLnPaymentRequest.getNumSatoshis());
+                                    showFeeAlertDialog(srb, feeLimitString);
+                                    return;
                                 }
```
Comment (michaelWuensch) [inline app/src/main/java/zapsolutions/zap/fragments/SendBSDFragment.java]: Is this reachable?

Comment (maraph) [reply app/src/main/java/zapsolutions/zap/fragments/SendBSDFragment.java]: `mLnFeePercentSettingLimit` will be 1 if the fee limit set to `None` (basically limit is 100%). Thats when we end up here and just do a "sanity check" if the fee would be higher than the payment as you mentioned in slack.

**Pull request merged**